payments-core/src/main/java/com/stripe/android/payments/core/injection/AuthenticationModule.kt

```diff
@@ -1,82 +1,39 @@
-import com.stripe.android.payments.core.authentication.NoOpIntentAuthenticator
-import com.stripe.android.payments.core.authentication.Stripe3DS2Authenticator
+import com.stripe.android.model.StripeIntent.NextActionData
+import com.stripe.android.payments.core.authentication.IntentAuthenticator
+import com.stripe.android.payments.core.authentication.OxxoAuthenticator
 import com.stripe.android.payments.core.authentication.WebIntentAuthenticator
-import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
-import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
-import com.stripe.android.view.AuthActivityStarterHost
+import dagger.Binds
 import dagger.Module
-import dagger.Provides
-import javax.inject.Singleton
-import kotlin.coroutines.CoroutineContext
+import dagger.multibindings.IntoMap
 
 /**
- * [Module] for com.stripe.android.payments.core.authentication.
- *
- * All non-daggerized dependencies are passed in as constructor parameters for now, a parameter
- * will be removed when it's daggerized and introduced to the dagger graph.
+ * Provides mappings between [NextActionData] and [IntentAuthenticator] provided by payment SDK.
  */
 @Module
-internal class AuthenticationModule(
-    private val stripeRepository: StripeRepository,
-    private val paymentRelayStarterFactory: (AuthActivityStarterHost) -> PaymentRelayStarter,
-    private val paymentBrowserAuthStarterFactory: (AuthActivityStarterHost) -> PaymentBrowserAuthStarter,
-    private val analyticsRequestExecutor: AnalyticsRequestExecutor,
-    private val analyticsRequestFactory: AnalyticsRequestFactory,
-    private val logger: Logger,
-    private val enableLogging: Boolean,
-    private val workContext: CoroutineContext,
-    private val uiContext: CoroutineContext,
-    private val threeDs2Service: StripeThreeDs2Service,
-    private val messageVersionRegistry: MessageVersionRegistry,
-    private val stripe3ds2Config: PaymentAuthConfig.Stripe3ds2Config,
-    private val stripe3ds2CompletionStarterFactory: (AuthActivityStarterHost, Int) -> Stripe3ds2CompletionStarter,
-) {
-    @Singleton
-    @Provides
-    internal fun provideNoOpAuthenticator(): NoOpIntentAuthenticator {
-        return NoOpIntentAuthenticator(paymentRelayStarterFactory)
-    }
+internal interface AuthenticationModule {
+    @IntentAuthenticatorMap
+    @Binds
+    @IntoMap
+    @IntentAuthenticatorKey(NextActionData.SdkData.Use3DS1::class)
+    fun binds3DS1Authenticator(webIntentAuthenticator: WebIntentAuthenticator): IntentAuthenticator
 
-    @Singleton
-    @Provides
-    internal fun provideWebAuthenticator(): WebIntentAuthenticator {
-        return WebIntentAuthenticator(
-            paymentBrowserAuthStarterFactory,
-            analyticsRequestExecutor,
-            analyticsRequestFactory,
-            logger,
-            enableLogging,
-            uiContext,
-        )
-    }
+    @IntentAuthenticatorMap
+    @Binds
```

Comment:
`AuthenticationModule` only binds already provided `IntentAuthenticator` instances to the map, it doesn't provide any instantiation